### PR TITLE
wxDataView optimizations.

### PR DESCRIPTION
**Pull request by erikhugintech**
As versatile as wxDataViewCtrl is, it falls short on large data sets, mainly due to inefficient use of sorting. This commit starts to address this.

Resort everything on thaw is removed. Recursive resorts are now applied only to open nodes. Node list sort state is retained, and node lists are only sorted when strictly necessary. Inserts into sorted lists are done at the correct position instead of appending and resorting. 

There should be ample comments in the diff, explaining rationales.

With this patch we're adding a few million items to a sorted wxDataViewCtrl without problematic performance. Without this patch, the same application pretty much froze for some hours while building the same item tree.

I've tried hard to follow your coding style guidelines, however as this style differs significantly from my personal style, it's quite possible that I've slipped here and there.

### CODE REVIEW

File: src/generic/datavgen.cpp [hunk 1]
```diff
@@ -2686,14 +2846,25 @@ bool wxDataViewMainWindow::ItemDeleted(const wxDataViewItem& parent,
...
+    }
 
     GetOwner()->InvalidateColBestWidths();
 
     // Send event
     wxDataViewEvent le(wxEVT_DATAVIEW_ITEM_VALUE_CHANGED, m_owner, item);
     m_owner->ProcessWindowEvent(le);
 
+	// Make sure the change is actually shown right away
+	UpdateDisplay();
```
Comment (vslavik) [inline src/generic/datavgen.cpp]: As someone looking into [noticeable performance problems even in *drawing code*](https://github.com/vslavik/poedit/issues/340), I’ll comment on this line: this, i.e. immediately redrawing *the entire control* on every change, would make already bad drawing performance even worse: even drawing the full control at idle time, as is currently done, is too slow.

Comment (erikhugintech) [inline src/generic/datavgen.cpp]: UpdateDisplay sets a flag to redraw the control later, it doesn't immediately update. I've not really looked into the drawing parts of the source though, I've simply noticed that value/item changes are not always reflected in the visible control unless I explicitly set m_dirty - aka UpdateDisplay()

Comment (vslavik) [inline src/generic/datavgen.cpp]: My apologies, I confused it with `RecalculateDisplay()`, mostly because I *knew* that every change causes idle-time repaint — that was a side-effect of always calling `Resort()`, though, so I think that got broken by your changes here. IOW, moving the `UpdateDisplay` call to be explicit here is the right thing to do, sorry for the noise.

**Comment by vslavik** [comment]
> Closed #444.

Why?

**Comment by erikhugintech** [comment]
Because I'm not very familiar with git, so I opened the pull request from my working branch assuming the pull would be from a specific commit, not a branch. I've reopened from a new branch.

**Comment by vslavik** [comment]
Ah, superseded by #459.

(I was worried you might have been discouraged by the lack of activity — don't, this PR is important, but everybody is super-busy and it *is* a complicated and thus hard(er) to review patch.)

**Comment by erikhugintech** [comment]
No worries - I'll modify what I need in my clone, and clean up and submit pull requests for what I think is usable for others than me, then it's all yours to do with as you wish. Figured I'd used wx for so many projects now it's time I contributed a little at least.
